### ISSUE TITLE
Twitterwall showing messages in wrong order

### ISSUE THREAD

**Issue by Orbiter** [loklak/loklak_webclient#154]
when I start a wall, the most recent tweets are shown first and then older tweets are shown.

**Comment by aneeshd16** [comment]
@Orbiter that was what I intended. After starting the wall, as soon as someone tweets, its given priority and shown earlier than other tweets. This gives the wall a *real time* effect. Can we have a discussion about the required functionality?

**Comment by Orbiter** [comment]
The wall actually has two phases:
- when it is first started, it shows all tweets which come from the search result
- when the wall reloads the next tweets, it shows all new tweets.

shure the wall shall be real-time, but that does not mean to show the tweets in the wrong order! why should the wall first show the most recent tweet and then older tweets?

**Comment by aneeshd16** [comment]
The search results are sorted by created_at. The most recent one is at the top. Currently the following happens:
* a sorted array of tweets is maintained (`nextStatuses`). The most recent tweet is the first tweet.
* The initial search results are sorted by default, so `nextStatuses` becomes the same as the search results array.
* Another function (`$interval`) gets the first tweet from `nextStatuses`, displays it and removes it from `nextStatuses` after every 4 seconds.
* Whenever a new search happens, the search results are added to `nextStatuses` (provided the tweet_id has not already been processed), and `nextStatuses` is sorted. In this way, whenever the `$interval` function runs, it will display the latest tweet, thus giving it real-time functionality.

In short, most recent tweets are shown first.

`shure the wall shall be real-time, but that does not mean to show the tweets in the wrong order! why should the wall first show the most recent tweet and then older tweets?`
Lets say the initial search results are as follows:
````
t1
t2
t3
t4
````
(`t1` is the latest tweet).
Therefore, `nextStatuses` will be same. ie:
````
t1
t2
t3
t4
````
The `$interval` functions displays t1 and removes it from `nextStatuses`. `nextStatuses` is now:
````
t2
t3
t4
````
At this point, a new search occurs and search results are:
````
t5
t1
t2
t3
t4
````
(`t5` is the newest tweet).
Thererfore, according to the logic `nextStatuses` will become:
````
t5
t2
t3
t4
````
(Since `t1` has already been displayed, it will not be shown again).
Now, `t5` gets shown before the other older tweets.
To test this, i did the following:
* Started the wall
* Tweeted with my account
* My tweet appeared on the wall within 30 seconds.

Sorry for the long post, but I believe getting the order of tweets right is important and I need to understand if this works or not :)

**Comment by mariobehling** [comment]
@aneeshd16 These are questions, we need to think about and test it. Maybe for the beginning we can just follow the design of existing walls. How does tweetwall.com handle this for example?

**Comment by aneeshd16** [comment]
@mariobehling , @Orbiter See this link http://kjsxs2fk.tweetwall.com/?preview=cats Tweetwall did the following:
* It showed the last three latest tweets.
* I tweeted, and my tweet appeared on the wall.
* It did not show older tweets.
* unless someone tweets, the tweetwall remains static.

**Comment by mariobehling** [comment]
@aneeshd16 I would say, we go with this for now and think about next steps. Tweetwall is just one example. I think the static option is not the final solution.

**Comment by aneeshd16** [comment]
Just clarifying, do you mean go with what we have currently or go with tweetwall?

**Comment by mariobehling** [comment]
Where can I test it?

**Comment by aneeshd16** [comment]
It should be available on almost all deployments. test.loklak.net works fine. This is the direct link to a created wall: http://test.loklak.net/wall/display?layoutStyle=1&headerColour=%2310405e&headerPosition=Top&headerForeColour=%23FFFFFF&eventName=myEvent&mainHashtagText=selfie&mainHashtag=%23loklak

**Comment by Orbiter** [comment]
arrr https://github.com/loklak/loklak_webclient/issues/154#issuecomment-111195310 the wall only shows tweets in the right order :ghost:

**Comment by aneeshd16** [comment]
I am still not sure about the order we need to follow. Our current wall or Tweetwall's? Thanks.

**Comment by mariobehling** [comment]
I think keep it with tweetwall.

**Comment by aneeshd16** [comment]
Solved in 2405abb4f9d4ce0f43d6a9a0d6f05a83493ce490
Will deploy and make it available for testing in the next PR after completing #71

**Comment by aneeshd16** [comment]
I believe this is solved. It now follows the same behavior as that of tweetwall.
